### PR TITLE
AI using View All to scout fog

### PR DESCRIPTION
**Pull request by idshibanov**

### CODE REVIEW

File: src/fheroes2/heroes/heroes_base.cpp [hunk 1]
```diff
@@ -339,23 +339,23 @@ bool HeroBase::CanCastSpell( const Spell & spell, std::string * res ) const
         return false;
     }
 
-    if ( !HaveSpell( spell ) ) {
+    if ( !HaveSpellPoints( spell ) ) {
```
Comment (oleg-derevenetz) [inline src/fheroes2/heroes/heroes_base.cpp:342]: Hi @idshibanov I just wonder, why the order of checks was changed here? Isn't it more logical to first check if the hero has a spell before checking the sufficiency of mana and movement points?

Comment (idshibanov) [reply src/fheroes2/heroes/heroes_base.cpp:342]: This is to just short-circuit the check. `HaveSpell` loops through both spells and artifacts while move/spell points checks should fail often too.

Comment (oleg-derevenetz) [reply src/fheroes2/heroes/heroes_base.cpp:342]: I hope this won't affect the spellcasting UI. I took a quick look - it doesn't seem like it should.

**Comment by oleg-derevenetz** [comment]
Hi @idshibanov could you please resolve the merge conflicts when you have free time so that we can proceed with this PR?

**Comment by oleg-derevenetz** [comment]
Hi @idshibanov Merged, thank you!

**Pull request merged**